### PR TITLE
refactor(promise): switch promise libraries

### LINKED ISSUE #68: Cannot do require('fxa-relier-client') using WebPack
Thanks for publishing the client to npm (https://github.com/mozilla/fxa-relier-client/issues/67) but we're unable to run the code using `require()` on the [payments-ui](https://github.com/mozilla/payments-ui) (which runs in a web browser). Is it possible to fix up the module for this?

Here is a simple case that reproduces the errors we see: https://dl.dropboxusercontent.com/u/254355/tmp/fxa-relier-packaging.zip

Unzip that directory, cd in and do this:
* `npm install`
* `grunt build`
* then open `index.html` in a web browser

My call to `require('fxa-relier-client')` results in `Error: Cannot find module '/client/FxaRelierClient.js'.`

I tried a couple things to work around the issue (e.g. adding resolve paths to `webpack.config.js`) but ultimately it leads to failures somewhere farther down the stack. I got as far as where `p-promise` was loading but failing because its test dependency `expect.js` wasn't there (which shouldn't be anyway).

Here are some reasons we'd like to use the client lib this way:
* we don't rely on bower anywhere in our stack
* we'd like to create a single minified bundle and using [webpack](http://webpack.github.io/) with common JS is the easiest way to do so
* it should be straight forward to fix up the module to support commonJS (I think)

Note also that similar errors occur if I install the client lib with bower and try to require the module from that directory.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #68

### CODE REVIEW

File: node/index.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ var path = require('path');
 var amd = require('./amd-loader');
 
 var map = {
-  'p-promise': 'p-promise'
+  'promise': 'es6-promise'
```
Comment (vladikoff) [inline node/index.js]: This part is causing issues at the moment. will need to adjust the node loader probably

**Comment by shane-tomlinson** [comment]
Possibly superseded by #77.

File: bower.json [hunk 2]
```diff
@@ -21,8 +21,9 @@
     "tests"
   ],
   "dependencies": {
-    "p-promise": "https://github.com/rkatic/p.git#3c1fe61c9ba47aa3c09a55eb3025bd29a238d1ea",
-    "micrajax": "0.0.5"
+    "p-promise": "0.5.0",
```
Comment (shane-tomlinson) [inline bower.json]: Should we ditch this now?

**Comment by shane-tomlinson** [comment]
Other than the extra `p-promise` include, if you are happy with this, I'm happy with this. r+ from me.

File: tests/extras/webpack.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        exec(BUNDLE_CMD, function (err, stdout, stderr) {
+          if (err) {
+            return dfd.reject(err);
+          }
+          assert.equal(stdout, 'ok\n');
+          return dfd.resolve();
+        });
+      })
```
Comment (shane-tomlinson) [inline tests/extras/webpack.js]: pedant alert - semicolon needed here. should we have eslint/jshint check for that?

Comment (pdehaan) [inline tests/extras/webpack.js]: FWIW, we don't seem to be linting the tests/** directory in this repo.
https://github.com/mozilla/fxa-relier-client/blob/master/tasks/eslint.js

Comment (pdehaan) [inline tests/extras/webpack.js]: I did a quick scan of the ESLint errors after adding the /tests/* directory and most were just single-vs-double quotes, or missing semicolons, or how it doesn't like strict mode.

But after tweaking all those locally, these were the only interesting errors:
```sh
$ grunt lint
Running "eslint:config" (eslint) task
Running "eslint:app" (eslint) task
Running "eslint:tests" (eslint) task

tests/spec/auth/api.js
   56:17  error  "p" is not defined                            no-undef
   79:10  error  configWithoutEmail is defined but never used  no-unused-vars
  119:17  error  "p" is not defined                            no-undef
  175:17  error  "p" is not defined                            no-undef
  232:17  error  "p" is not defined                            no-undef

✖ 5 problems (5 errors, 0 warnings)
```

**Pull request closed without merging**